addons/l10n_in_edi/views/account_move_views.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="invoice_form_inherit_l10n_in_edi" model="ir.ui.view">
+        <field name="name">account.move.form.inherit.l10n.in.edi</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//group[@name='sale_info_group']" position="inside">
+                <field name="l10n_in_edi_cancel_reason" attrs="{'invisible': ['|', ('country_code', '!=', 'IN'), ('state', '!=', 'posted')]}"/>
```

Comment:
It only makes sense showing it when the document is at least sent (or to_cancel or cancelled)

Reply:
okay